src/sources/GeoJSON.js

```diff
@@ -386,18 +391,22 @@ export default class GeoJSON extends Base {
             clipped: []
         };
         let holeIndex = 0;
+        let firstReverse = false;
         for (let i = 0; i < data.length; i++) {
-            for (let j = 0; j < data[i].length; j++) {
-                const point = this._computePointGeometry(data[i][j]);
-                polygon.flat.push(point.x, point.y);
+            let reverse = this._isReversed(data[i]);
+            if (i === 0) {
+                firstReverse = reverse;
```

Comment:
We can add here a pre condition, to set the first reverse before the loop and start with `i = 1` to avoid this check in every loop.